src/create_update.c

```diff
@@ -385,16 +392,17 @@ int main(int argc, char **argv)
 		/* Detect renamed files specifically for os-core */
 		rename_detection(new_core);
 #endif
-		old_deleted = remove_old_deleted_files(old_core, new_core);
+		old_deleted = remove_deprecated_files(old_core, new_core, comp_deleted);
+		old_ghosted = remove_deprecated_files(old_core, new_core, comp_ghosted);
 		sort_manifest_by_version(new_core); /* sorts by filename */
 		newfiles = prune_manifest(new_core);
 		if (newfiles <= 0) {
 			LOG(NULL, "", "Core component has not changed (after pruning), exiting");
 			printf("Core component has not changed (after pruning), exiting\n");
 			goto exit;
 		}
-		LOG(NULL, "", "Core component has changes (%d new, %d deleted), writing out new manifest", newfiles, old_deleted);
-		printf("Core component has changes (%d new, %d deleted), writing out new manifest\n", newfiles, old_deleted);
+		LOG(NULL, "", "Core component has changes (%d new, %d deleted, %d ghosted), writing out new manifest", newfiles, old_deleted, old_ghosted);
```

Comment:
I agree it would be nice to have a single function, but as @busykai pointed out  the duplicate `LOG` and `printf` calls happen elsewhere in the code as well. They should all be fixed up together.